File: 2231_digit-generator/2231.cpp
#include <iostream>
#include <string>

using namespace std;

int main(void)
{
  // freopen("example.txt", "r", stdin);
  int input;
  cin >> input;

  for (int i = 0; i < input; i++)
  {
    int sum = i;

    int num = i;
    while (num != 0)
    {
      sum += num % 10;
      num /= 10;
    }

    if (sum == input)
    {
      cout << i;
      return 0;
    }
  }

  cout << 0;
  return 0;
}

File: 2231_digit-generator/2231.js
const input = require('fs')
  .readFileSync(
    process.platform === 'linux'
      ? '/dev/stdin'
      : require('path').resolve(__dirname, 'example.txt')
  )
  .toString()
  .trim();

const N = parseInt(input);

let M = 0;
for (let i = 0; i < N; i++) {
  let sum = i;

  const stringValue = i.toString();
  for (let j = 0; j < stringValue.length; j++) {
    sum += parseInt(stringValue[j]);
  }

  if (sum === N) {
    M = i;
    break;
  }
}

console.log(M);
